juju/juju
# Worker V4 package update

Brings in the worker v4 package, this incudes the changes to the start function
on the manifold config. The new start function now offers a context.Context.
This context.Context is a context for only the duration of the start function,
not the entire life of the worker. This context is used to cancel the start
function if the worker is shutting down.

There might be a possibility in the future to have a Fork function that allows 
the passing of values of a context, that negates the canceling of the
context.

Lastly, I need to fix the dependency test package to update the old naming of
context. That can be done in a separate PR.

## Checklist




## QA steps

Quick sanity check:

```
$ juju bootstrap lxd test --build-agent
$ juju enable-ha
$ juju deploy ubuntu
```

## Links

**Jira card:** JUJU-5286
